src/.chezmoiexternal.toml.tmpl

```diff
@@ -39,4 +39,20 @@ type = "archive-file"
 url = "https://github.com/ajeetdsouza/zoxide/releases/download/v{{ .zoxide.version }}/zoxide-{{ .zoxide.version }}-{{ if eq .chezmoi.os "darwin" }}{{ if eq .chezmoi.arch "arm64" }}aarch64{{ else }}x86_64{{ end }}-apple-darwin{{ else }}{{ if eq .chezmoi.arch "arm64" }}aarch64{{ else }}x86_64{{ end }}-unknown-linux-musl{{ end }}.tar.gz"
 path = "zoxide"
 executable = true
-{{ end }}
\ No newline at end of file
+{{ end }}
+
+{{ if eq .zsh.oh-my-zsh.installation "external-sources" }}
+[".oh-my-zsh"]
+    type = "archive"
+    url = "https://github.com/ohmyzsh/ohmyzsh/archive/{{ .zsh.oh-my-zsh.sha }}.tar.gz"
+    exact = true
+    stripComponents = 1
+{{ end }}
```

Comment:
preference here is to use `with` to scope to the the `.zsh.oh-my-zsh` settings in a single block.